node_modules/@angular/cli/commands/add-impl.d.ts(7,37): error TS2307: Cannot find module '../../../../dist-schema/packages/angular/cli/lib/config/schema'.

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 bug report

### Affected Package

The issue seems to be caused by `@angular/bazel`.

### Description

This error is popin : `node_modules/@angular/cli/commands/add-impl.d.ts(7,37): error TS2307: Cannot find module '../../../../dist-schema/packages/angular/cli/lib/config/schema'. `

But it seems that  it does exist.

```
ls -al node_modules/@angular/cli/lib/config/schema.*
-rw-r--r-- 1 beclood becloodusers  6044 Oct 26  1985 node_modules/@angular/cli/lib/config/schema.d.ts
-rw-r--r-- 1 beclood becloodusers  1748 Oct 26  1985 node_modules/@angular/cli/lib/config/schema.js
-rw-r--r-- 1 beclood becloodusers 68891 Oct 26  1985 node_modules/@angular/cli/lib/config/schema.json
```

Pretty same issue here: https://github.com/angular/angular/issues/33411

Just install a fresh environment on Docker with `debian:buster`.
```
# NodeJS and NPM
RUN wget https://nodejs.org/dist/v12.13.0/node-v12.13.0-linux-x64.tar.xz
RUN tar --strip-components 1 -xf node-v12.13.0-linux-x64.tar.xz -C /usr/local

# Angular config
RUN npm install --global @angular/cli 
RUN npm install --global @angular/bazel
```

## 🔬 Minimal Reproduction

Just launched the command from the bazel guide : https://angular.io/guide/bazel

## 🔥 Exception or Error

<pre><code>
ng new --collection=@angular/bazel
? What name would you like to use for the new workspace and initial project? qcm
? Would you like to add Angular routing? Yes
? Which stylesheet format would you like to use? SCSS   [ http://sass-lang.com/documentation/file.SASS_REFERENCE.html#syntax ]
CREATE qcm/README.md (1021 bytes)
CREATE qcm/.editorconfig (246 bytes)
CREATE qcm/.gitignore (631 bytes)
CREATE qcm/angular.json (2135 bytes)
CREATE qcm/package.json (1547 bytes)
CREATE qcm/tsconfig.json (543 bytes)
CREATE qcm/tslint.json (1953 bytes)
CREATE qcm/browserslist (429 bytes)
CREATE qcm/karma.conf.js (1015 bytes)
CREATE qcm/tsconfig.app.json (270 bytes)
CREATE qcm/tsconfig.spec.json (270 bytes)
CREATE qcm/src/favicon.ico (948 bytes)
CREATE qcm/src/index.html (289 bytes)
CREATE qcm/src/main.ts (372 bytes)
CREATE qcm/src/polyfills.ts (2838 bytes)
CREATE qcm/src/styles.scss (80 bytes)
CREATE qcm/src/test.ts (642 bytes)
CREATE qcm/src/assets/.gitkeep (0 bytes)
CREATE qcm/src/environments/environment.prod.ts (51 bytes)
CREATE qcm/src/environments/environment.ts (662 bytes)
CREATE qcm/src/app/app-routing.module.ts (246 bytes)
CREATE qcm/src/app/app.module.ts (393 bytes)
CREATE qcm/src/app/app.component.scss (0 bytes)
CREATE qcm/src/app/app.component.html (25530 bytes)
CREATE qcm/src/app/app.component.spec.ts (1089 bytes)
CREATE qcm/src/app/app.component.ts (208 bytes)
CREATE qcm/e2e/protractor.conf.js (808 bytes)
CREATE qcm/e2e/tsconfig.json (214 bytes)
CREATE qcm/e2e/src/app.e2e-spec.ts (636 bytes)
CREATE qcm/e2e/src/app.po.ts (262 bytes)
CREATE qcm/angular-metadata.tsconfig.json (729 bytes)
CREATE qcm/e2e/protractor.on-prepare.js (1112 bytes)
CREATE qcm/src/initialize_testbed.ts (432 bytes)
CREATE qcm/src/main.dev.ts (185 bytes)
CREATE qcm/src/main.prod.ts (249 bytes)
CREATE qcm/src/rxjs_shims.js (1259 bytes)
CREATE qcm/angular.json.bak (3791 bytes)

> @bazel/hide-bazel-files@0.39.1 postinstall /app/qcm/node_modules/@bazel/hide-bazel-files
> node ./index.js


> core-js@2.6.10 postinstall /app/qcm/node_modules/babel-runtime/node_modules/core-js
> node postinstall || echo "ignore"


> core-js@3.2.1 postinstall /app/qcm/node_modules/core-js
> node scripts/postinstall || echo "ignore"


> core-js@2.6.10 postinstall /app/qcm/node_modules/karma/node_modules/core-js
> node postinstall || echo "ignore"


> protobufjs@6.8.8 postinstall /app/qcm/node_modules/protobufjs
> node scripts/postinstall


> sauce-connect-launcher@1.3.1 postinstall /app/qcm/node_modules/sauce-connect-launcher
> node scripts/install.js || nodejs scripts/install.js


> @angular/cli@8.3.16 postinstall /app/qcm/node_modules/@angular/cli
> node ./bin/postinstall/script.js


> qcm@0.0.0 postinstall /app/qcm
> ngc -p ./angular-metadata.tsconfig.json

node_modules/@angular/cli/commands/add-impl.d.ts(7,37): error TS2307: Cannot find module '../../../../dist-schema/packages/angular/cli/lib/config/schema'.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! qcm@0.0.0 postinstall: `ngc -p ./angular-metadata.tsconfig.json`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the qcm@0.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/beclood/.npm/_logs/2019-10-31T08_37_49_179Z-debug.log
Package install failed, see above.
</code></pre>


## 🌍  Your Environment

**Angular Version:**
<pre><code>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.3.15
Node: 12.13.0
OS: linux x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.803.15
@angular-devkit/core         8.3.15
@angular-devkit/schematics   8.3.15
@schematics/angular          8.3.15
@schematics/update           0.803.15
rxjs                         6.4.0
</code></pre>

--- a/integration/bazel/angular-metadata.tsconfig.json
+++ b/integration/bazel/angular-metadata.tsconfig.json
@@ -19,6 +19,7 @@
     "node_modules/@angular/**/*"
   ],
   "exclude": [
+    "node_modules/@angular/cli/**",
     "node_modules/@angular/bazel/**",
     "node_modules/@angular/**/testing/**",
     "node_modules/@angular/**/schematics/**",

--- a/packages/bazel/src/schematics/ng-add/files/angular-metadata.tsconfig.json.template
+++ b/packages/bazel/src/schematics/ng-add/files/angular-metadata.tsconfig.json.template
@@ -16,6 +16,7 @@
     "node_modules/@angular/**/*"
   ],
   "exclude": [
+    "node_modules/@angular/cli/**",
     "node_modules/@angular/bazel/**",
     "node_modules/@angular/**/schematics/**",
     "node_modules/@angular/**/testing/**",

